File: Problem1.java
package Misc-5;
// TC: O(n)
// SC: O(n)
public class Problem1 {
    public int longestOnes(int[] nums, int k) {
        int slow=0;
        int max=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==0) k--;
            if(k<0){
                if(nums[slow] == 0){
                    k++;
                }
                slow++;
            }
        }

        return nums.length - slow;
    }
}

File: problem1.cpp
#include<iostream>
#include<vector>
#include<math.h>

using namespace std;

class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int sz = nums.size();
        int count{},max_len{},slow{};
        for(int i{};i<sz;++i){
            if(nums.at(i)==0){
                count++;
            }
            if(count>k){
                max_len = max(max_len,i-slow);
                while(nums.at(slow)!=0){
                    slow++;
                }
                slow++;
                count--;
            }
        }
        max_len = max(max_len,sz-slow);
        return max_len;
    }
};

class Solution_2 {
public:
    int longestOnes(vector<int>& nums, int k) {
        int sz = nums.size();
        int slow{},count{};
        for(int i{};i<sz;++i){
            if(nums.at(i)==0){
                count++;
            }
            if(count>k){
                if(nums.at(slow)==0){
                    count--;
                }
                slow++;
            }
        }
        return sz-slow;
    }
};

